core/src/main/java/tc/oc/pgm/filters/FilterMatchModule.java

```diff
@@ -530,7 +535,9 @@ public void onPartyChange(PlayerPartyChangeEvent event) throws EventException {
                 }
               });
 
-      event.yield();
+      if (BukkitUtils.isSportPaper()) {
+        event.yield(); // TODO: what does not having this, when using spigot / paper, break?
```

Comment:
That sounds like it would require some more significant reworking. Could I just simply disable dynamic filters when a server isn't running SportPaper?

Reply:
I'd reqlly advise against that idea, its going to break half the maps. Thats just a non-functioning pgm you're getting

Reply:
Ah yeah, I see what you mean from the [docs](https://pgm.dev/docs/modules/mechanics/filters#player-filters) about how many filters are dynamic. Ideally, I'd like to avoid refactoring dynamic filters in this PR. Do you have any suggestions for a strategy for this?

Reply:
I think you're gonna need to search for work-arounds even if they're ugly (reflection/nms) for these. I dont think there's a better way around it.

Reply:
The idea behind a NoOp NMS is basically as a fallback for when NMS doesn't work. ie bukkit only, no craftbukkit

Reply:
Why are we evwn trying to supoort that? Pgm fundamentqlly relies on craftbukkit/nms. What's the target we expect to run on?

Reply:
Well really in this case the target is all pgm versions between 1.8 and 1.20. A NoOp implementation in general would make it easier to implement specific version implementations. Specifically, for my purposes, NMSHacksNoOp is targeting only bukkit so that I can increment the Minecraft version number and make PGM able to run with the changes introduced in each version, without having to upgrade an entire NMSHacks implementation. I can then get each change properly reviewed and allow future PRs to be smaller, and therefore less likely to miss any issues. I believe that if we left this on a separate branch and just kept adding changes until we have a version that supports 1.8 and 1.20 that it simply wouldn't happen, and with that approach we will never have any sort of multi-version support.

Reply:
Also, given how many features do work with no NMS in this PR, I don't think PGM is as tied to craftbukkit/nms as you think. It is definitely a lesser experience, but not as much as you might think.

Reply:
If the no-op impl.os only to be used as a base, and it itself cant really function, why not make it abstract and leave methods which cannot be reasonably implemented within bukkit api unimplemented, forcing children (which can have some reliance on CF/nms) to implement those?

Reply:
> given how many features do work with no NMS in this PR, I don't think PGM is as tied to craftbukkit/nms as you think

I think the 80/20 rule applies here, where 80% of functionality can be achieved with 20% of the work, and the remaining 20% of the functionality takes 80% of the work.
PGM will "mostly" work on a generic bukkit implementation (non-craftbukkit) but there will be the 20% of missing/broken/subtly-broken features, or bugs that will hide there and no one will fix. I'd rather not claim we support non-craftbukkit bukkit API implementations, and just plainly tell such users on startuo we don't support whatever thing they're trying to run on

Reply:
I'm good with keeping it abstract. I agree that we don't want people to try and run the NoOp implementation thinking it is properly supported.

Reply:
Added a cleanup at the beginning of the next tick for when the platform don't support yielding